src/core/namespace.c

```diff
@@ -724,6 +796,27 @@ static int mount_empty_dir(const MountEntry *m) {
         return 1;
 }
 
+static int mount_tmpfs(const MountEntry *m) {
+        unsigned long flags = MS_NODEV|MS_STRICTATIME;
+        int r;
+
+        assert(m);
+
+        /* First, get rid of everything that is below if there is anything. Then, overmount with our new empty dir */
+
+        (void) mkdir_p_label(mount_entry_path(m), 0755);
+        (void) umount_recursive(mount_entry_path(m), 0);
+
+        r = mnt_optstr_get_flags(mount_entry_option(m), &flags, mnt_get_builtin_optmap(MNT_LINUX_MAP));
+        if (r < 0)
+                return log_debug_errno(r, "Failed to get mount flags from mount options %s: %m", mount_entry_option(m));
+
+        if (mount("tmpfs", mount_entry_path(m), "tmpfs", flags, mount_entry_option(m)) < 0)
```

Comment:
I'm confused :(.

Clearly your tests are passing, and you made sure to try and exercise the libmount function.  But you're still passing *all* the text options to the mount() system call, right?  When I asked if that was ok, it said no...

```
$ cat > test2.c
#include <sys/mount.h>

int main() {
        mount("tmp", "/mnt", "tmpfs", MS_NODEV|MS_STRICTATIME|MS_RDONLY, "ro");
        mount("tmp", "/mnt", "tmpfs", MS_NODEV|MS_STRICTATIME, "");
        mount("tmp", "/mnt", "tmpfs", MS_NODEV|MS_STRICTATIME|MS_RDONLY, "");
        mount("tmp", "/mnt", "tmpfs", MS_RDONLY, "ro");
}
$ make test2 >/dev/null
$ sudo strace -e trace=mount test2
mount("tmp", "/mnt", "tmpfs", MS_RDONLY|MS_NODEV|MS_STRICTATIME, "ro") = -1 EINVAL (Invalid argument)
mount("tmp", "/mnt", "tmpfs", MS_NODEV|MS_STRICTATIME, "") = 0
mount("tmp", "/mnt", "tmpfs", MS_RDONLY|MS_NODEV|MS_STRICTATIME, "") = 0
mount("tmp", "/mnt", "tmpfs", MS_RDONLY, "ro") = -1 EINVAL (Invalid argument)
```

Do you know why your tests are passing, and my paranoia was not valid?

Reply:
In principle it would also be nice if the tests exercised one of the non-flag options as well.

Reply:
Thank you for the example. In the updated version, all flags are removed from the option string.